proto/message.py

```diff
@@ -159,6 +158,15 @@ def __new__(mcls, name, bases, attrs):
             # Increment the field index counter.
             index += 1
 
+        # As per descriptor.proto, all synthetic oneofs must be ordered after
+        # 'real' oneofs.
+        for field in fields:
+            if field.optional:
+                field.oneof = "_{}".format(field.name)
+                field.descriptor.oneof_index = oneofs[field.oneof] = len(oneofs)
+                fget = lambda self: self.__class__.pb(self).HasField(field.name)
+                attrs['Has_{}'.format(field.name)] = property(fget, None)
```

Comment:
It seems like logic similar to this could be added to `__contains__`, which I think is the right thing to do.

Reply:
Changed the interface; this code no longer exists.